es-app/src/guis/GuiGamelistOptions.cpp

```diff
@@ -298,7 +298,8 @@ GuiGamelistOptions::GuiGamelistOptions(Window* window, IGameListView* gamelist,
 			auto srcSystem = file->getSourceFileData()->getSystem();
 			auto sysOptions = mSystem->isGroupSystem() ? srcSystem : mSystem;
 
-			bool showSystemOptions = ApiSystem::getInstance()->isScriptingSupported(ApiSystem::GAMESETTINGS) && (sysOptions->hasFeatures() || sysOptions->hasEmulatorSelection());
+			//bool showSystemOptions = ApiSystem::getInstance()->isScriptingSupported(ApiSystem::GAMESETTINGS) && (sysOptions->hasFeatures() || sysOptions->hasEmulatorSelection());

```

Comment:
find a better way to do this, I cannot accept this as its disabling features for ALL of the systems.

Reply:
It's not disabling the features of the systems it's just always presenting the Advanced System Option tab so the resolution option can always be present along with any system features if there is any. The other option I suppose would be to move the native Resolution change into it's own category under Advanced system options, but that might be overkill. Let me know, cheers.

Reply:
That is what es_features.cfg is for. I have some free time today, all we need to do is create the feature that we need (in this case nativevideo) and add it to the emulator that needs it.

Reply:
try with this patch

```
diff --git a/es-app/src/SystemData.cpp b/es-app/src/SystemData.cpp
index 997cca89..67277fc1 100644
--- a/es-app/src/SystemData.cpp
+++ b/es-app/src/SystemData.cpp
@@ -482,6 +482,7 @@ EmulatorFeatures::Features EmulatorFeatures::parseFeatures(const std::string fea
 		if (trim == "autocontrollers") ret = ret | EmulatorFeatures::Features::autocontrollers;
 #ifdef _ENABLEEMUELEC
 		if (trim == "vertical") ret = ret | EmulatorFeatures::Features::vertical;		
+		if (trim == "nativevideo") ret = ret | EmulatorFeatures::Features::nativevideo;		
 #endif
 	}
 
diff --git a/es-app/src/SystemData.h b/es-app/src/SystemData.h
index ab8798a2..0881770b 100644
--- a/es-app/src/SystemData.h
+++ b/es-app/src/SystemData.h
@@ -75,6 +75,7 @@ public:
 		autocontrollers = 262144,
 #ifdef _ENABLEEMUELEC
         vertical = 524288,
+        nativevideo = 1048576,
 #endif
 
 		all = 0x0FFFFFFF
diff --git a/es-app/src/guis/GuiMenu.cpp b/es-app/src/guis/GuiMenu.cpp
index 61329970..4a678499 100644
--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -4340,6 +4340,17 @@ void GuiMenu::popSpecificConfigurationGui(Window* mWindow, std::string title, st
 
 		systemConfiguration->addWithLabel(_("Emulator"), emulChoice);
 	}
+#ifdef _ENABLEEMUELEC
+    if (systemData->isFeatureSupported(currentEmulator, currentCore, EmulatorFeatures::nativevideo))
+	{
+        auto videoNativeResolutionMode_choice = createNativeVideoResolutionModeOptionList(mWindow, configName);
+        systemConfiguration->addWithLabel(_("NATIVE VIDEO"), videoNativeResolutionMode_choice);
+        systemConfiguration->addSaveFunc([configName, videoNativeResolutionMode_choice] {
+            SystemConf::getInstance()->set(configName + ".nativevideo", videoNativeResolutionMode_choice->getSelected());
+            SystemConf::getInstance()->saveSystemConf();
+        });
+    }
+#endif
 
 	// Screen ratio choice
 	if (systemData->isFeatureSupported(currentEmulator, currentCore, EmulatorFeatures::ratio))
@@ -4819,6 +4830,36 @@ std::shared_ptr<OptionListComponent<std::string>> GuiMenu::createRatioOptionList
 	return ratio_choice;
 }
 
+#ifdef _ENABLEEMUELEC
+std::shared_ptr<OptionListComponent<std::string>> GuiMenu::createNativeVideoResolutionModeOptionList(Window *window, std::string configname)
+{
+	auto emuelec_video_mode = std::make_shared< OptionListComponent<std::string> >(window, "NATIVE VIDEO", false);
+	std::vector<std::string> videomode;
+	videomode.push_back("1080p60hz");
+	videomode.push_back("1080i60hz");
+	videomode.push_back("720p60hz");
+	videomode.push_back("720p50hz");
+	videomode.push_back("480p60hz");
+	videomode.push_back("480cvbs");
+	videomode.push_back("576p50hz");
+	videomode.push_back("1080p50hz");
+	videomode.push_back("1080i50hz");
+	videomode.push_back("576cvbs");
+
+	for (auto it = videomode.cbegin(); it != videomode.cend(); it++) {
+		std::string index = SystemConf::getInstance()->get(configname + ".nativevideo");
+		if (index.empty())
+			index = SystemConf::getInstance()->get("global.videomode");
+		if (index.empty())
+			index = SystemConf::getInstance()->get("ee_videomode");
+
+		emuelec_video_mode->add(*it, *it, index == *it);
+	}
+
+	return emuelec_video_mode;
+}
+#endif
+
 std::shared_ptr<OptionListComponent<std::string>> GuiMenu::createVideoResolutionModeOptionList(Window *window, std::string configname) 
 {
 	auto videoResolutionMode_choice = std::make_shared<OptionListComponent<std::string> >(window, _("VIDEO MODE"), false);
diff --git a/es-app/src/guis/GuiMenu.h b/es-app/src/guis/GuiMenu.h
index 93151a71..8f36719d 100644
--- a/es-app/src/guis/GuiMenu.h
+++ b/es-app/src/guis/GuiMenu.h
@@ -93,6 +93,9 @@ private:
 	TextComponent mVersion;
 
 	static std::shared_ptr<OptionListComponent<std::string>> createRatioOptionList(Window *window, std::string configname);
+#ifdef _ENABLEEMUELEC
+	static std::shared_ptr<OptionListComponent<std::string>> createNativeVideoResolutionModeOptionList(Window *window, std::string configname);
+#endif
 	static std::shared_ptr<OptionListComponent<std::string>> createVideoResolutionModeOptionList(Window *window, std::string configname);
 	static void popSpecificConfigurationGui(Window* mWindow, std::string title, std::string configName, SystemData *systemData, FileData* fileData, bool selectCoreLine = false);
```

And add `<emulator name="dolphin" features="nativevideo" />` to `es_features.cfg` (or whichever emulator needs this feature)

IMHO this is the correct way of doing it. Also notice that I have added some `#ifdef _ENABLEEMUELEC` because I like to mark  the EmuELEC changes.